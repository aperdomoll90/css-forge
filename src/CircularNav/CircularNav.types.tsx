export interface CircularNavPropsTypes {
  MenuItemsArray: MenuItemsArrayPropsTypes[]
  color?: string
  hoverColor?: string
  pressColor?: string
}

export interface MenuItemsArrayPropsTypes {
  label?: string
  icon: React.SVGProps<SVGSVGElement>
  link: string
}
