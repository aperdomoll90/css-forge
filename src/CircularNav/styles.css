.circularNavMenu {
  width: 15rem;
  height: 4rem;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: 0.5s ease-in-out;
}

.circularNavActive {
  height: 15rem;
}

.circularNavMenu li {
  position: absolute;
  left: 0;
  list-style: none;
  font-size: 1.6em;
  transform-origin: 7.5rem;
  transition: 0.5s;
  transition-delay: calc(0.1s * var(--i));
  transform: rotate(0deg) translateX(6rem);
}

.circularNavMenu.circularNavActive li {
  transform: rotate(calc(360deg / 8 * var(--i)));
}

.circularNavToggle {
  width: 3.5rem;
  height: 3.5rem;
  position: absolute;
  border-radius: 50%;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 100;
  background: #fff;
  box-shadow: 0 3px 4px rgba(0, 0, 0, 0.15);
  font-size: 3em;
  transition: 1.25s;
}
.circularNavToggle:hover {
  background: rgb(155, 153, 153);
}

.circularNavMenu.circularNavActive .circularNavToggle {
  transform: rotate(315deg);
}

.circularNavMenu li a {
  width: 2.5rem;
  height: 2.5rem;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #fff;
  transition: 0.5s;
  box-shadow: 0 3px 4px rgba(0, 0, 0, 0.15);
  transform: rotate(calc(360deg / -8 * var(--i)));
}
.circularNavMenu li a:hover {
  background: rgba(155, 153, 153, 0.856);
}

/*  Iconography */
.circularNavToggle-icon {
  height: 80%;
  width: 80%;
}
.circularNavMenu-icon svg {
  height: 70%;
  width: 70%;
}

.circularNavToggle-icon path,
.circularNavMenu-icon path,
.circularNavMenu-icon rect,
.circularNavMenu-icon circle {
  stroke-width: 40;
  stroke: var(--color);
  stroke-linecap: round;
  stroke-linejoin: round;
  fill: transparent;
}

.circularNavToggle-icon:hover path,
.circularNavMenu-icon:hover path,
.circularNavToggle-icon:hover rect,
.circularNavToggle-icon:hover circle {
  stroke: var(--hoverColor);
}

.circularNavToggle-icon:target path,
.circularNavMenu-icon:target path,
.circularNavToggle-icon:target rect,
.circularNavToggle-icon:target circle {
  stroke: var(--pressColor);
}

@media (max-width: 850px) {
  .circularNavActive {
    height: 15rem;
  }
}
