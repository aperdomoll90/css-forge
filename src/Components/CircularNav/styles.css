.circularNavWrapper {
  position: absolute;
  top: 10px;
  left: calc(50% - 7.5rem);
  z-index: 1000;
}

.circularNavMenu {
  width: 15rem;
  height: 4rem;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: 0.5s ease-in-out;
}

.circularNavActive {
  height: 15rem;
}

@media (max-width: 850px) {
  .circularNavWrapper {
    top: -20px;
    left: calc(50% - 7.5rem);
    transform: scale(80%);
  }
  .circularNavActive {
    height: 15rem;
  }
}
@media (max-width: 500px) {
  .circularNavWrapper {
    transform: scale(90%);
  }
}

.circularNavMenu li {
  position: absolute;
  left: 0;
  list-style: none;
  font-size: 1.6em;
  color: #fff;
  transform-origin: 7.5rem;
  transition: 0.5s;
  transition-delay: calc(0.1s * var(--i));
  transform: rotate(0deg) translateX(6rem);
}

.circularNavMenu.circularNavActive li {
  transform: rotate(calc(360deg / 8 * var(--i)));
}


.circularNavToggle {
  width: 3.5rem;
  height: 3.5rem;
  position: absolute;
  border-radius: 50%;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 100;
  background: #fff;
  box-shadow: 0 3px 4px rgba(0, 0, 0, 0.15);
  font-size: 3em;
  color: #000;
  transition: 1.25s;
}
.circularNavToggle:hover {
  color: darkblue;
  background: rgb(155, 153, 153);
}

.circularNavMenu.circularNavActive .circularNavToggle {
  transform: rotate(315deg);
}

.circularNavMenu li a {
  width: 2.5rem;
  height: 2.5rem;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #fff;
  color: #000;
  transition: 0.5s;
  box-shadow: 0 3px 4px rgba(0, 0, 0, 0.15);
  transform: rotate(calc(360deg / -8 * var(--i)));

}
.circularNavMenu li a:hover {
  color: rgb(63, 63, 214);
  background: rgba(155, 153, 153, 0.856);
}
