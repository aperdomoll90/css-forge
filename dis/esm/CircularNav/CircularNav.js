var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, Fragment as _Fragment, jsxs as _jsxs } from "react/jsx-runtime";
import { useEffect } from 'react';
import './styles.css';
import '../utils/GlobalStyles.css';
import { AddIcon } from '../IconCollection/IconCollection';
export var CircularNav = function (_a) {
    var menuItemsArray = _a.menuItemsArray, color = _a.color, hoverColor = _a.hoverColor, pressColor = _a.pressColor;
    useEffect(function () {
        var toggle = document.querySelector('.circularNavToggle');
        var menu = document.querySelector('.circularNavMenu');
        toggle.onclick = function () {
            menu.classList.toggle('circularNavActive');
        };
    }, []);
    return (_jsx("div", __assign({ className: 'circularNavWrapper' }, { children: _jsxs("div", __assign({ className: 'circularNavMenu', style: { '--color': color, '--hoverColor': hoverColor, '--pressColor': pressColor } }, { children: [_jsx("div", __assign({ className: 'circularNavToggle' }, { children: _jsx("div", __assign({ className: 'circularNavToggle-icon' }, { children: AddIcon })) })), menuItemsArray &&
                    menuItemsArray.map(function (item, index) {
                        return (_jsx("li", __assign({ className: 'circularNavMenu-icon', style: { '--i': index } }, { children: _jsx("a", __assign({ href: item.link, className: 'circularNavMenu-icon' }, { children: _jsx(_Fragment, { children: item.icon }) })) })));
                    })] })) })));
};
//# sourceMappingURL=CircularNav.js.map