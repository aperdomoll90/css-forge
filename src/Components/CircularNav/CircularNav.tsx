import { useEffect } from 'react';
import { Meta, Title, IconGallery, IconItem } from '@storybook/blocks';

import { Icon as IconExample } from './Icon';
import './styles.css'


function CircularNav() {
    
    useEffect(() => {
        let toggle = document.querySelector('.circularNavToggle') as HTMLElement;
        let menu = document.querySelector('.circularNavMenu') as HTMLElement
        toggle.onclick = () => { menu.classList.toggle('circularNavActive')}
    }, [])

  return (   
 <div className="circularNavWrapper">
    <div className="circularNavMenu">
           <div className="circularNavToggle"><ion-icon name="add-outline"/></div>
           <li style={{'--i':0} as React.CSSProperties}><a href='#'><ion-icon name="home-outline"/></a></li>
           <li style={{'--i':1} as React.CSSProperties}><a href='#'><ion-icon name="person-outline"/></a></li>
           <li style={{'--i':2} as React.CSSProperties}><a href='#'><ion-icon name="settings-outline"/></a></li>
           <li style={{'--i':3} as React.CSSProperties}><a href='#'><ion-icon name="mail-outline"/></a></li>
           <li style={{'--i':4} as React.CSSProperties}><a href='#'><ion-icon name="key-outline"/></a></li>
           <li style={{'--i':5} as React.CSSProperties}><a href='#'><ion-icon name="videocam-outline"/></a></li>
           <li style={{'--i':6} as React.CSSProperties}><a href='#'><ion-icon name="water-outline"/></a></li>
           <li style={{'--i':7} as React.CSSProperties}><a href='#'><ion-icon name="paw-outline"/></a></li>
                
     </div>
</div>
  )
}

export default CircularNav
