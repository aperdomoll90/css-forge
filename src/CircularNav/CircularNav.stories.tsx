import { Meta, StoryFn } from '@storybook/react'
import { CircularNav } from './CircularNav'
import { AtomIcon, BarIcon, CameraIcon, CodeIcon, GearIcon, HomeIcon, MapIcon, PawIcon, PlanetIcon, ProfileIcon } from '../IconCollection/IconCollection'
import StorybookContainer from '../StorybookContainer'
import { CircularNavPropsTypes, MenuItemsArrayPropsTypes } from './CircularNav.types'

export default {
  title: 'Navigation Bars',
  component: CircularNav,
  parameters: {
    jest: ['CircularNav.test.tsx'],
  },
  argTypes: {
    color: { options: ['red', 'black', 'yellow', 'purple'], control: { type: 'radio' } },
    hoverColor: { options: ['red', 'black', 'yellow', 'purple'], control: { type: 'radio' } },
    pressColor: { options: ['red', 'black', 'yellow', 'purple'], control: { type: 'radio' } },
  },
} as Meta

const MenuItemsArray: MenuItemsArrayPropsTypes[] = [
  {
    label: 'Home',
    icon: HomeIcon,
    link: '#',
  },
  {
    label: 'Settings',
    icon: GearIcon,
    link: '#',
  },
  {
    label: 'Map',
    icon: MapIcon,
    link: '#',
  },
  {
    label: 'Analytics',
    icon: BarIcon,
    link: '#',
  },
  {
    label: 'Animations',
    icon: CameraIcon,
    link: '#',
  },
  {
    label: 'Profile',
    icon: ProfileIcon,
    link: '#',
  },
  {
    label: 'Pets',
    icon: PawIcon,
    link: '#',
  },
  {
    label: 'Code',
    icon: CodeIcon,
    link: '#',
  },
]

const CircularNavTemplate: StoryFn<CircularNavPropsTypes> = args => (
  <StorybookContainer>
    <div style={{ position: 'absolute', top: '20%', left: 'calc(50% - 7.5rem)' }}>
      <CircularNav {...args} />
    </div>
  </StorybookContainer>
)

export const CircularNavComponent = CircularNavTemplate.bind({})
CircularNavComponent.args = {
  MenuItemsArray: MenuItemsArray,
  color: 'rgb(28, 28, 28)',
  hoverColor: 'red',
  pressColor: 'yellow',
}
