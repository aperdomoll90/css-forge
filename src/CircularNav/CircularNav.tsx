import { useEffect } from 'react'
import './styles.css'
import { AddIcon } from '../IconCollection/IconCollection'
import { CircularNavPropsTypes, MenuItemsArrayPropsTypes } from './CircularNav.types'

export const CircularNav: React.FC<CircularNavPropsTypes> = ({ MenuItemsArray, color, hoverColor, pressColor }) => {
  useEffect(() => {
    let toggle = document.querySelector('.circularNavToggle') as HTMLElement
    let menu = document.querySelector('.circularNavMenu') as HTMLElement
    toggle.onclick = () => {
      menu.classList.toggle('circularNavActive')
    }
  }, [])

  return (
    <div className='circularNavWrapper'>
      <div className='circularNavMenu' style={{ '--color': color, '--hoverColor': hoverColor, '--pressColor': pressColor } as React.CSSProperties}>
        <div className='circularNavToggle'>
          <div className='circularNavToggle-icon'>{AddIcon}</div>
        </div>
        {MenuItemsArray &&
          MenuItemsArray.map((item: MenuItemsArrayPropsTypes, index: number) => {
            return (
              <li className='circularNavMenu-icon' style={{ '--i': index } as React.CSSProperties}>
                <a href={item.link} className='circularNavMenu-icon'>
                  {item.icon}
                </a>
              </li>
            )
          })}
      </div>
    </div>
  )
}
