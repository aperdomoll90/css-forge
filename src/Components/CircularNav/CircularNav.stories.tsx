import { Meta, StoryFn } from '@storybook/react'
import CircularNav from './CircularNav'
import StorybookContainer from '../StorybookContainer'

export default {
  title: 'Navigation Bars',
  component: CircularNav,
  parameters: {
    jest: ['CircularNav.test.tsx'],
  },
} as Meta

const CircularNavTemplate: StoryFn = () => (
  <StorybookContainer>
    <CircularNav />
  </StorybookContainer>
)

export const CircularNavComponent = CircularNavTemplate.bind({})
