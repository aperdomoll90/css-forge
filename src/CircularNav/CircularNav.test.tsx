import { render } from '@testing-library/react'
import { CircularNav } from './CircularNav'

describe('CircularNav component', () => {
  const renderComponent = () => render(<CircularNav />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
