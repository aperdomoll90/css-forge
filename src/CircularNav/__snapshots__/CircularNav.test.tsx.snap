// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularNav component should match snapshots 1`] = `
<div>
  <div
    class="circularNavWrapper"
  >
    <div
      class="circularNavMenu"
    >
      <div
        class="circularNavToggle"
      >
        <div
          class="circularNavToggle-icon"
        >
          <svg
            id="Icon-Add"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Add
            </title>
            <path
              class="icon-add"
              d="M256 112v288M400 256H112"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
